apps/bootloader.js

```diff
@@ -101,12 +101,28 @@ if (startapp) {
   function drawWidgets() {
     for (var w of WIDGETS) w.draw();
   }
-  var clockApps = require("Storage").list().filter(a=>a[0]=='+').map(app=>{
-    try { return require("Storage").readJSON(app); }
-    catch (e) {}
-  }).filter(app=>app.type=="clock").sort((a, b) => a.sortorder - b.sortorder);
-  if (clockApps && clockApps.length > 0) eval(require("Storage").read(clockApps[0].src));
+  var settings;
+  try {
+    settings = require("Storage").readJSON('@setting');
+  } catch (e) {
+    settings = {}
+  }
+  var clockApp = settings.clock;
+  if (clockApp) {
+      clockApp = require("Storage").read(clockApp)
```

Comment:
What happens when you have some app as default in settings, but you deleted it?

Reply:
If it has been deleted then `clockApp` will be `undefined` after line 112, so the check on line 114 kicks in and we go back to loading the first clock app we can find.